lib/src/functions/math.dart

```diff
@@ -133,6 +133,46 @@ final _hypot = BuiltInCallable("hypot", r"$numbers...", (arguments) {
       numeratorUnits: numeratorUnits, denominatorUnits: denominatorUnits);
 });
 
+///
+/// Exponential functions
+///
+
+final _log = BuiltInCallable("log", r"$number, $base: null", (arguments) {
+  var number = arguments[0].assertNumber("number");
+  if (arguments[1] == sassNull) {
+    number.assertNoUnits();
+    return SassNumber(math.log(number.value));
+  }
+  var base = arguments[1].assertNumber("base");
+  if (number.hasUnits) {
+    throw SassScriptException("Expected \$number to have no units.");
```

Comment:
If you format these exceptions like `$argname: message`, they'll match auto-generated exceptions and get highlighted nicely once we implement https://github.com/sass/dart-sass/issues/793.

Reply:
Done, thanks for the context.